templates/consul-maint-before-shutdown.service.j2

```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=Consul maintenance task at shutdown
+After=consul.service
+After={{ consul_service_name }}.service
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+ExecStop=/usr/local/bin/consul maint -enable
+ExecStopPost=/usr/bin/sleep 30
```

Comment:
This duration should be templated too. I would recommend a higher timeout, e.g. 45 seconds or 60 seconds, to avoid race conditions between DNS cache expiration and DNS client resolution that may happen simultaneously, or with a few seconds of delay.

Also, a higher value may require adding a `TimeoutStopSec=` (default 90 seconds).

This duration should only be that high for publicly exposed instances, such as HAProxies or Coturn services (maybe others?). Other internal services are resolved on-the-spot via the Consul resolver, and could do with a lower timeout, e.g. 5 or 10 seconds.

Do you know what happens if we `systemctl restart haproxy` for example? Do it wait for 30 seconds too?

Reply:
For the `TimeoutStopSec` option, do you think we will happen to need more than 90 seconds ? Otherwise it stops the service (with an error for sure)

Reply:
If `systemctl restart haproxy` , nothing happens, it still waits

Reply:
About the `TimeoutStopSec` option, it's only needed if we template the sleep duration. If it is templated, either document to not exceed 90seconds, or template the `TimeoutStopSec` as well.

About the restart, does it mean that running `systemctl restart haproxy` will enable Consul maintenance, wait 30 seconds, restart HAProxy, and leave Consul maintenance enabled? So restarting HAProxy would remove the instance from traffic?